### DISCUSSION TITLE
Option to use different storage backend other than Local Disk

### DISCUSSION THREAD

**Discussion by alkmim** [immich-app/immich#3385] [General]
Hi

I started deploying Immich at home a couple of days ago. I really got surprised with the features it has so far and the amount of contributors.

Currently, Immich is using local storage to keep its data (please, let me know if I am wrong). I was wondering about adding a feature to support multiple storage backends instead of just the local disk. 

One storage backend that would be handy is S3 compatible.  

Please, let me know your thoughts.

**Comment by alextran1502** [comment]
I think you can already do this with rclone

**Reply by alkmim** [reply]
I believe in order to make it work with Immich we would have to use the rclone mount feature (https://rclone.org/commands/rclone_mount/), which uses FUSE. 

From my experience, fuse can considerable impact performance and there are some limitation (not sure how they impact Immich). For the limitations, see: https://rclone.org/commands/rclone_mount/#limitations

Another downside of it is that setting it up on kubernetes will not be as straightforward as change a couple of configuration parameters.

Anyway, I will give it a try and update here about how it works.

**Comment by bo0tzz** [comment]
Duplicate of #1683
